website/src/components/SearchPage/fields/DateRangeField.tsx

```diff
@@ -0,0 +1,134 @@
+    }
+}
+
+export const DateRangeField = ({ field, fieldValues, setSomeFieldValues }: DateRangeFieldProps) => {
+    const lowerFromField = field.groupedFields.filter(
+        (f) => f.name.endsWith('From') && f.rangeOverlapSearch!.bound === 'lower',
+    )[0];
+    const lowerToField = field.groupedFields.filter(
+        (f) => f.name.endsWith('To') && f.rangeOverlapSearch!.bound === 'lower',
+    )[0];
+    const upperFromField = field.groupedFields.filter(
+        (f) => f.name.endsWith('From') && f.rangeOverlapSearch!.bound === 'upper',
+    )[0];
+    const upperToField = field.groupedFields.filter(
+        (f) => f.name.endsWith('To') && f.rangeOverlapSearch!.bound === 'upper',
+    )[0];
+    // TODO maybe all these '!' should be handled differently?
+
+    const [strictMode, setStrictMode] = useState(
+        isStrictMode(
+            lowerFromField.name in fieldValues,
+            lowerToField.name in fieldValues,
+            upperFromField.name in fieldValues,
+            upperToField.name in fieldValues,
+        ),
+    );
+
+    const lowerField = strictMode ? lowerFromField : upperFromField;
+    const upperField = strictMode ? upperToField : lowerToField;
+
+    const [lowerValue, setLowerValue] = useState(fieldValues[lowerField.name] ?? '');
+    const [upperValue, setUpperValue] = useState(fieldValues[upperField.name] ?? '');
+
+    useEffect(() => {
+        if (strictMode) {
+            setSomeFieldValues({
+                [lowerFromField.name]: lowerValue,
+                [upperToField.name]: upperValue,
+                [upperFromField.name]: null,
+                [lowerToField.name]: null,
+            });
+        } else {
+            setSomeFieldValues({
+                [upperFromField.name]: lowerValue,
+                [lowerToField.name]: upperValue,
+                [lowerFromField.name]: null,
+                [upperToField.name]: null,
+            });
+        }
+    }, [
+        strictMode,
+        lowerValue,
+        upperValue,
+        lowerFromField,
+        lowerToField,
+        upperFromField,
+        upperToField,
+        setSomeFieldValues,
+    ]);
+
+    return (
+        <div key={field.name} className='flex flex-col border p-3 mb-3 rounded-md border-gray-300'>
+            <div className='flex flex-row justify-between items-baseline mb-2'>
+                <h3 className='text-gray-500 text-sm'>{field.displayName}</h3>
+                <CustomTooltip id={'strict-tooltip' + field.name}>
+                    <div className='w-52'>{TOOLTIP_TEXT}</div>
+                </CustomTooltip>
+                <label data-tooltip-id={'strict-tooltip' + field.name}>
+                    <span className='text-gray-400 text-sm mr-2'>strict</span>
+                    <input
+                        type='checkbox'
+                        className='checkbox checkbox-sm [--chkbg:theme(colors.gray.400)]'
```

Comment:
> the checkbox looks a bit like it's disabled, but with more solid colors it draws a lot of attention. But making it smaller would also not be great for clicking it I think. 🤔 Haven't come up with a good solution for this yet. (but maybe not critical, I don't think it looks terrible)

I don't think it looks terrible and I don't think we need to block, but I imagine it could be better with white bg and gray tick rather than the reverse

Reply:
That's a great idea! I implemented it in https://github.com/loculus-project/loculus/pull/3308/commits/5d5ea4f354871414fc4b513fd1f29957e46c6e2c (also updated the screenshots)